language/stdlib/modules/LibraAccount.move

```diff
@@ -1051,126 +1060,105 @@ module LibraAccount {
-        assert(exists_at(transaction_sender), PROLOGUE_EACCOUNT_DNE);
+        assert(exists_at(transaction_sender), Errors::invalid_argument(PROLOGUE_EACCOUNT_DNE));
 
         // We check whether this account is frozen, if it is no transaction can be sent from it.
         assert(
             !AccountFreezing::account_is_frozen(transaction_sender),
-            PROLOGUE_EACCOUNT_FROZEN
+            Errors::invalid_state(PROLOGUE_EACCOUNT_FROZEN)
         );
 
         // Load the transaction sender's account
-        let sender_account = borrow_global_mut<LibraAccount>(transaction_sender);
+        let sender_account = borrow_global<LibraAccount>(transaction_sender);
 
         // Check that the hash of the transaction's public key matches the account's auth key
         assert(
             Hash::sha3_256(txn_public_key) == *&sender_account.authentication_key,
-            PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY
+            Errors::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY),
         );
 
         // Check that the account has enough balance for all of the gas
         assert(
             (txn_gas_price as u128) * (txn_max_gas_units as u128) <= MAX_U64,
-             PROLOGUE_ECANT_PAY_GAS_DEPOSIT
+            Errors::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT),
         );
         let max_transaction_fee = txn_gas_price * txn_max_gas_units;
         // Don't grab the balance if the transaction fee is zero
         if (max_transaction_fee > 0) {
             let balance_amount = balance<Token>(transaction_sender);
-            assert(balance_amount >= max_transaction_fee, PROLOGUE_ECANT_PAY_GAS_DEPOSIT);
+            assert(
+                balance_amount >= max_transaction_fee,
+                Errors::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
+            );
         };
 
         // Check that the transaction sequence number matches the sequence number of the account
-        // TODO: the below assertions overlap, fix this.
         assert(
             txn_sequence_number >= sender_account.sequence_number,
-            PROLOGUE_ESEQUENCE_NUMBER_TOO_OLD
+            Errors::invalid_argument(PROLOGUE_ESEQUENCE_NUMBER_TOO_OLD)
         );
         assert(
             txn_sequence_number == sender_account.sequence_number,
-            PROLOGUE_ESEQUENCE_NUMBER_TOO_NEW
+            Errors::invalid_argument(PROLOGUE_ESEQUENCE_NUMBER_TOO_NEW)
         );
         assert(
             LibraTimestamp::now_seconds() < txn_expiration_time_seconds,
-            PROLOGUE_ETRANSACTION_EXPIRED
+            Errors::invalid_argument(PROLOGUE_ETRANSACTION_EXPIRED)
         );
     }
 
-    /// Collects gas and bumps the sequence number for executing a transaction
+    /// Collects gas and bumps the sequence number for executing a transaction.
+    /// The epilogue is invoked at the end of the transaction.
+    /// If the exection of the epilogue fails, it is re-invoked with different arguments guaranteed
```

Comment:
I hesitate a bit at the word "guaranteed". Maybe hedge a bit and phrase it as "should never"?